packages/wallets/src/api/client.ts

```diff
@@ -0,0 +1,302 @@
+import { validateAPIKey, environmentToCrossmintBaseURL, APIKeyUsageOrigin } from "@crossmint/common-sdk-base";
+import { type Client, createClient, createConfig } from "@hey-api/client-fetch";
+
+import type {
+    ClientOptions,
+    CreateWalletDto,
+    WalletV1Alpha2ResponseDto,
+    CreateTransactionDto,
+    SubmitApprovalDto,
+    WalletsV1ControllerGetTransaction4Response,
+    WalletsV1Alpha2TransactionResponseDto,
+    CreateSignatureRequestDto,
+    WalletsV1Alpha2SignatureResponseDto,
+    WalletsV1Alpha2TransactionsResponseDto,
+    Nftevm,
+    Nftsol,
+    WalletBalanceResponseDto,
+} from "./gen/types.gen";
+import {
+    walletsV1ControllerCreateWallet4,
+    walletsV1ControllerGetWalletByLocator4,
+    walletsV1ControllerCreateTransaction4,
+    walletsV1ControllerSubmitApprovals4,
+    walletsV1ControllerGetTransaction4,
+    walletsV1ControllerCreateSignatureRequest4,
+    walletsV1ControllerSubmitSignatureApprovals4,
+    walletsV1ControllerGetSignature4,
+    walletsV1ControllerGetTransactionsWithoutChain4,
+    fetchContentFromWallet,
+    balanceControllerGetBalanceForLocator2,
+} from "./gen/sdk.gen";
+
+type CreateWalletParams = CreateWalletDto;
+type CreateWalletResponse = WalletV1Alpha2ResponseDto;
+type GetWalletResponse = WalletV1Alpha2ResponseDto;
+
+type CreateTransactionParams = CreateTransactionDto;
+type CreateTransactionResponse = WalletsV1Alpha2TransactionResponseDto;
+type ApproveTransactionParams = SubmitApprovalDto;
+type ApproveTransactionResponse = WalletsV1ControllerGetTransaction4Response;
+type GetTransactionResponse = WalletsV1Alpha2TransactionResponseDto;
+
+type CreateSignatureParams = CreateSignatureRequestDto;
+type CreateSignatureResponse = WalletsV1Alpha2SignatureResponseDto;
+type ApproveSignatureParams = SubmitApprovalDto;
+type ApproveSignatureResponse = WalletsV1Alpha2SignatureResponseDto;
+type GetSignatureResponse = WalletsV1Alpha2SignatureResponseDto;
+
+type GetTransactionsResponse = WalletsV1Alpha2TransactionsResponseDto;
+type GetNftsResponse = Nftevm | Nftsol;
+type GetBalanceResponse = WalletBalanceResponseDto;
+
+class ApiClient {
+    private readonly client: Client;
+
+    constructor(
+        private readonly apiKey: string,
+        private readonly jwt?: string
+    ) {
+        const validationResult = validateAPIKey(this.apiKey);
+        if (!validationResult.isValid) {
+            throw new Error("Invalid API key");
+        }
+        if (validationResult.usageOrigin === APIKeyUsageOrigin.CLIENT && !jwt) {
+            throw new Error("JWT is required for client API key");
```

Comment:
Im nt sure jwt is always needed - why did you reach that conclusion?

Reply:
I think it depends on the specific API you;re calling

Reply:
We check the presence of JWT when the API key is client-side. 

Are there any situations where we access the client-side APIs, but don't have the JWT?

Reply:
Yes! Anything that doesn't hit /me I think

For example i think fetching wallet balance and others

In general - we dont have an invariant in the server that is "if its a client side api, it requires jwt". It is something we configure call by call